File: app/src/main/java/com/tech/dagger2_android/inJava/network/NetworkClient.java
package com.tech.dagger2_android.inJava.network;

import android.util.Log;

import com.tech.dagger2_android.inJava.utils.Constants;

import javax.inject.Inject;

public class NetworkClient {
     @Inject
     NetworkConnection mConnection;

    @Inject
    public NetworkClient(NetworkConnection networkConnection){
        mConnection = networkConnection;
    }

    //Method Injection
    @Inject
    public void testMethod(){
        Log.d(Constants.TAG,"testMethod : This is a test Method");
    }
    @Inject   //Method Injection
    public void attachedToNetworkTester(NetworkTester tester){
        tester.attachToNetwork(this);
    }
    public String fetchData(){
        return mConnection.doReq();
    }
}

File: app/src/main/java/com/tech/dagger2_android/inKotlin/network/NetworkClient.kt
package com.tech.dagger2_android.inKotlin.network

import javax.inject.Inject

class NetworkClient @Inject constructor(networkConnection : NetworkConnection?) {

    private var mConnection: NetworkConnection? = null

    init {
        mConnection = networkConnection
    }
    fun fetchData(): String? {
        return mConnection?.doReq()
    }
}